### ISSUE TITLE
fela overrides styles after rehydration

### ISSUE THREAD

**Issue by vlad-zhukov** [rofrischmann/fela#405]
**Version**: 6.0.x
**Environment**: Both
**Type**: Bug

-------------------
#### Description
For example I've got a button with the following styles:
```js
{
  color: 'white',
  ':hover': {
    backgroundColor: 'black',
  },
}
```
that will be transformed into:
```css
.a {color: white}
.b:hover {backgroundColor: black}
```

I render this button on server into a stylesheet, then on client I render this button, then another element, for example, an `h1` with `color: 'red'`, and the resulting stylesheet will be:
```css
.a {color: white}
.b:hover {backgroundColor: black}
.b {color: red}
```
which means my button will have a red color.

**Comment by lostpebble** [comment]
Yep, on the new version of Fela seeing similar issues with hydration.

For example, I have a footer who's styles render last. But for some reason the footer's background colour will overwrite the background colour of another earlier created rule. I've reverted to the old versions to get things working properly again.

I can't pinpoint exactly where this goes wrong. When I removed the background colour from the footer rule, it then moved on to the next style in the footer, which was `height: 10em` and applied that to the earlier created rule.
